src/Storages/ExternalStream/Kafka/WriteBufferFromKafka.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <atomic>
+#include <librdkafka/rdkafka.h>
+
+#include "IO/BufferWithOwnMemory.h"
+#include "IO/WriteBuffer.h"
+
+namespace DB
+{
+
+using RdKafkaTopicPtr = std::unique_ptr<rd_kafka_topic_t, void (*)(rd_kafka_topic_t *)>;
+
+class WriteBufferFromKafka : public BufferWithOwnMemory<WriteBuffer>
+{
+private:
+    struct State
+    {
+        std::atomic_size_t outstandings;
```

Comment:
let's be explicit the width of a variable if it is need serde (serialized / deserialized) like `std::atomic_uint64_t or std::atomic<uint64_t>` etc

Reply:
`State` just needs to stay in memory and no serde is required ( e.g. it's not a checkpoint state ).